### PR TITLE
regeln anlegen funktioniert nicht, fixes #360

### LINKED ISSUE #360: Regeln anlegen funktioniert nicht
Man kann aktuell keine Regeln anlegen. Wenn man eine Zeit angibt (zB. 120) dann wird diese als Fehler markiert: "Muss größer als 0 sein" und das Feld ist gefüllt mit: -345669

### PR DESCRIPTION
**Pull request by bitboxer**
@moonglum @bascht would you please check if this is now working correct for you?

Thanks for @bascht to pinpointing the exact problem.

### CODE REVIEW

**Comment by bitboxer** [comment]
@moonglum can you look into this, too? This is the important one I want to have on production asap to get RheinRuhr startet <3

**Comment by moonglum** [comment]
Currently creating an event with recurring dates looks like this on the hacken.in Vagrant box for me (With nothing done but `rake db:seed`). 

screen shot 2015-04-08 at 20 07 50

**Comment by moonglum** [comment]
Not sure if this is because of your fix or not, will investigate tomorrow.

**Comment by moonglum** [comment]
What the actual fuck is going on here... Creating it on the console gives me the same error...

```
PG::UniqueViolation: ERROR:  duplicate key value violates unique constraint "events_pkey"
DETAIL:  Key (id)=(4) already exists.
: INSERT INTO "events" ("category_id", "created_at", "name", "schedule_yaml", "updated_at") VALUES ($1, $2, $3, $4, $5) RETURNING "id"
   (0.3ms)  ROLLBACK
ActiveRecord::RecordNotUnique: PG::UniqueViolation: ERROR:  duplicate key value violates unique constraint "events_pkey"
DETAIL:  Key (id)=(4) already exists.
: INSERT INTO "events" ("category_id", "created_at", "name", "schedule_yaml", "updated_at") VALUES ($1, $2, $3, $4, $5) RETURNING "id"
from /home/vagrant/.rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/activerecord-4.1.8/lib/active_record/connection_adapters/postgresql_adapter.rb:834:in `get_last_result'
```

**Comment by moonglum** [comment]
In the VM I have insufficient privileges to drop the DB, @bascht (when executing `rake db:drop`). We need to check that.

**Comment by moonglum** [comment]
So I can't get into Postgres with `psql` on the Vagrant Box to throw away the box (and can't do it with Rails' Rake Tasks as described above). Little bit stuck here :cry: What I gathered from [here](https://github.com/hacken-in/puppet-hackenin/blob/master/manifests/postgresql.pp):

* The username of the user I'm supposed to use is "user". The password is "password".
* The root password is "password"

And still can't login. Maybe my limited understanding of Puppet is the problem here. Halp, @bascht. Will continue tomorrow after some feedback :heart:

**Comment by bitboxer** [comment]
Ping @moonglum

**Comment by bascht** [comment]
Let me have a look.

**Comment by bascht** [comment]
Alright. Didn't have time to check it until now. This isn't exactly a Vagrant / Puppet issue. It's plain ol' PostgreSQL. There is no such thing as a database `root` user, but you can switch users, using the system `root` user (see example). I think you mixed up the `postgresql` unix user / database user. (The unix user has no password set).

### Switching users
```sql
vagrant@hacken:/opt/hacken.in$ sudo su - postgres
postgres@hacken:~$ psql
psql (9.3.6)
Type "help" for help.

postgres=# SELECT VERSION();
                                                  version                                                   
------------------------------------------------------------------------------------------------------------
 PostgreSQL 9.3.6 on x86_64-unknown-linux-gnu, compiled by gcc (Ubuntu/Linaro 4.6.3-1ubuntu5) 4.6.3, 64-bit
(1 row)
```

### Using privileges

All databases inside of the VM are created by the user `postgres` - so that's close to using root with mysql. :) On the other side you shouldn't run into problems then. Check your `config/databases.yml`. We could still run into problems if you're creating databases from one of the base templates. I'd suggest we create a different template (with correct encoding) to use with the hacken.in database - just to be sure. Using a template it would look like this.

#### drop / create
```yaml
development:
  adapter: postgresql
  database: hcking
  username: postgres
  host: localhost
  password: password
  template: hcking_template

# Warning: The database defined as "test" will be erased and
# re-generated from your development database when you run "rake".
# Do not set this db to the same as development or production.
test:
  adapter: postgresql
  database: hcking_test
  username: postgres
  host: localhost
  password: password
  template: hcking_template
```
```shell
vagrant@hacken:/opt/hacken.in$ sudo su postgres -c "psql -c '\l'" 
                                     List of databases
      Name       |  Owner   | Encoding |   Collate   |    Ctype    |   Access privileges   
-----------------+----------+----------+-------------+-------------+-----------------------
 hcking          | postgres | UTF8     | en_US.UTF-8 | en_US.UTF-8 | 
 hcking_template | postgres | UTF8     | en_US.UTF-8 | en_US.UTF-8 | =T/postgres          +
                 |          |          |             |             | postgres=CTc/postgres+
                 |          |          |             |             | user=CTc/postgres
 hcking_test     | postgres | UTF8     | en_US.UTF-8 | en_US.UTF-8 | 
 postgres        | postgres | LATIN1   | en_US       | en_US       | 
 template0       | postgres | LATIN1   | en_US       | en_US       | =c/postgres          +
                 |          |          |             |             | postgres=CTc/postgres
 template1       | postgres | LATIN1   | en_US       | en_US       | =c/postgres          +
                 |          |          |             |             | postgres=CTc/postgres
vagrant@hacken:/opt/hacken.in$ bundle exec rake db:drop
vagrant@hacken:/opt/hacken.in$ sudo su postgres -c "psql -c '\l'"                                                                                                                                                                                                                                 
                                     List of databases
      Name       |  Owner   | Encoding |   Collate   |    Ctype    |   Access privileges   
-----------------+----------+----------+-------------+-------------+-----------------------
 hcking_template | postgres | UTF8     | en_US.UTF-8 | en_US.UTF-8 | =T/postgres          +
                 |          |          |             |             | postgres=CTc/postgres+
                 |          |          |             |             | user=CTc/postgres
 postgres        | postgres | LATIN1   | en_US       | en_US       | 
 template0       | postgres | LATIN1   | en_US       | en_US       | =c/postgres          +
                 |          |          |             |             | postgres=CTc/postgres
 template1       | postgres | LATIN1   | en_US       | en_US       | =c/postgres          +
                 |          |          |             |             | postgres=CTc/postgres
(4 rows)

vagrant@hacken:/opt/hacken.in$ bundle exec rake db:create
vagrant@hacken:/opt/hacken.in$ sudo su postgres -c "psql -c '\l'" 
                                     List of databases
      Name       |  Owner   | Encoding |   Collate   |    Ctype    |   Access privileges   
-----------------+----------+----------+-------------+-------------+-----------------------
 hcking          | postgres | UTF8     | en_US.UTF-8 | en_US.UTF-8 | 
 hcking_template | postgres | UTF8     | en_US.UTF-8 | en_US.UTF-8 | =T/postgres          +
                 |          |          |             |             | postgres=CTc/postgres+
                 |          |          |             |             | user=CTc/postgres
 hcking_test     | postgres | UTF8     | en_US.UTF-8 | en_US.UTF-8 | 
 postgres        | postgres | LATIN1   | en_US       | en_US       | 
 template0       | postgres | LATIN1   | en_US       | en_US       | =c/postgres          +
                 |          |          |             |             | postgres=CTc/postgres
 template1       | postgres | LATIN1   | en_US       | en_US       | =c/postgres          +
                 |          |          |             |             | postgres=CTc/postgres
(6 rows)
```

I'm currently polishing up the commits so I can introduce it into the puppet module, but the permission problem @moonglum saw, is not a bug. I repeat: It's not a bug. :-)

**Comment by bascht** [comment]
To indulge in the goodness of the aforementioned changes, just rebase onto #580. :fireworks:

File: spec/models/event_spec.rb [hunk 1]
```diff
@@ -365,4 +365,21 @@ def create_week_based_event(time)
 
     gevent.destroy
   end
+
+  describe 'duration wandering away bug #360' do
+    before do
+      @event = FactoryGirl.create(:simple, schedule_yaml: "---\n:start_date: 2011-08-07 13:13:00.000000000 +02:00\n:duration: 10800\n:rrules: []\n:exrules: []\n:rtimes:\n- 2011-08-08 19:00:00.000000000 +02:00\n- 2011-11-15 19:00:00.000000000 +01:00\n- 2012-02-06 19:00:00.000000000 +01:00\n:extimes: []\n")
+      @current_zone = Time.zone
+      Time.zone = ActiveSupport::TimeZone["Vancoover"]
```
Comment (bascht) [inline spec/models/event_spec.rb]: I think this spec is bogus. I can change it to `ActiveSupport::TimeZone["Rosinenbrötchen"]` or delete the complete line. Doesn't change the outcome of the test.

**Comment by bascht** [comment]
@bitboxer The test case is bogus (see comment) but the functionality is fine. Your fix is working and once the test case is fixed, I'd merge this. Thanks for looking into it! :+1:

**Comment by moonglum** [comment]
The magical @bascht. Thanks for pointing that out to me, will try it :+1:

File: spec/models/event_spec.rb (hunk 1 above)
Comment (moonglum) [inline spec/models/event_spec.rb]: I'm not sure if your deduction that the spec is bogus is correct. It is also possible that this line is superfluous :smile: I will try if without this line the spec is failing without the changes in the other file.

**Comment by moonglum** [comment]
I still get the duplicate error, even after dropping and recreating the database... Want to merge #580 first.

**Comment by moonglum** [comment]
Rebased.

**Comment by moonglum** [comment]
Both @bascht and I ran into this error. It was not due to this PR though, but an error in our seeds.

**Comment by moonglum** [comment]
Thanks to all of you! Happy this is finally merged :+1:

**Pull request merged**